pkg/internal/ebpf/common/pids.go

```diff
@@ -18,24 +18,24 @@ const (
 	PIDTypeGo
 )
 
-var activePids, _ = lru.New[uint32, svc.ID](1024)
+var activePids, _ = lru.New[uint32, *svc.ID](1024)
```

Comment:
I'm thinking about reverting this one, it works but it's fragile. Essentially we give back the pointer for usage by the code that builds the eBPF pid filtering. If that code ever changed to modify the SVC, it might break things. It should be minimal allocation impact, since we only need that code when we add and remove pids.

Reply:
I see that this map is only used with system-wide optimization.